src/helpers.ts

```diff
@@ -0,0 +1,203 @@
+
+    return result.visitorId;
+  }
+
+  protected static async getPayload(
+    projectApiKey: string,
+    userId: number | string,
+    eventType: EventType,
+    eventOptions?: EventOptions,
+  ): Promise<TrackPayload> {
+    const visitorId = await RecureAIClientHelper.getOrSetVisitorId('visitorId', projectApiKey);
+
+    return {
+      userId: userId.toString(),
+      provider: 'recure-pixel',
+      type: eventType,
+      visitorId,
+      timestamp: new Date().toISOString(),
+      eventOptions,
+    };
+  }
+
+  protected static isReadyToSend(): boolean {
+    const cReadyToSend: string | undefined = Cookies.get(sendingEventsBlocked);
+    const lSReadyToSend: string | null = localStorage.getItem(sendingEventsBlocked);
+
+    const expirationTime: Date = new Date(new Date().getTime() + minutesToExpire * toMinutes * toSeconds);
+
+    if (!cReadyToSend && !lSReadyToSend) {
+      RecureAIClientHelper.setExpirationDate(expirationTime);
+
+      return true;
+    }
+
+    if (!cReadyToSend && lSReadyToSend) {
+      const eventTimeGap: number = RecureAIClientHelper.getDifferenceBetweenCurrentAndLocalDateEvent(lSReadyToSend);
+
+      if (eventTimeGap > 0) {
+        RecureAIClientHelper.setExpirationDate(expirationTime);
+
+        return true;
+      } else {
+        RecureAIClientHelper.setCookieWhenDateExistInLocalStorage(lSReadyToSend);
+
+        return false;
+      }
+    }
+
+    return false;
+  }
+
+  private static createFilterPath(filter: FilterOptions): string {
+    const filters = {
+      flags: filter?.flags?.join(','),
+      types: filter?.types?.join(','),
+      from: filter?.from?.toISOString(),
+      to: filter?.to?.toISOString(),
+      page: filter?.page,
+    };
+
+    return qs.stringify(filters);
+  }
+
+  protected static async getAccountsDetails(projectApiKey: string, path: ApiPath, apiParams?: ApiParams): Promise<any> {
+    let apiPath = accountApiUrl;
+
+    if (path === ApiPath.ACCOUNTS && apiParams?.filter) {
+      apiPath += `?${RecureAIClientHelper.createFilterPath(apiParams.filter)}`;
+    } else if (path === ApiPath.DETAIL_ACCOUNT) {
+      apiPath += `${apiParams?.accountDetails?.accountId}/`;
+    } else if (path === ApiPath.DETAIL_ACCOUNT_EVENTS) {
+      apiPath += `${apiParams?.accountDetails?.accountId}/events/`;
+      if (apiParams?.filter) {
+        apiPath += `?${RecureAIClientHelper.createFilterPath(apiParams.filter)}`;
+      }
+    } else if (path === ApiPath.DETAIL_ACCOUNT_DETAIL_EVENTS) {
+      apiPath += `${apiParams?.accountDetails?.accountId}/events/${apiParams?.accountDetails?.eventId}/`;
+    } else if (path === ApiPath.DETAIL_ACCOUNT_FLAGS) {
+      apiPath += `${apiParams?.accountDetails?.accountId}/flags/`;
+    } else if (path === ApiPath.DETAIL_ACCOUNT_DETAIL_FLAGS) {
+      apiPath += `${apiParams?.accountDetails?.accountId}/flags/${apiParams?.accountDetails?.flagId}/`;
+    }
+
+    try {
+      return await fetch(apiPath, {
```

Comment:
```suggestion
      return fetch(apiPath, {
```
you don't need await here, you anyway return promise

Reply:
fixed